src/ORM/Association/BelongsToMany.php

```diff
@@ -157,6 +164,52 @@ class BelongsToMany extends Association
     protected $_sort;
 
     /**
+     * Sets the name of the field representing the binding key to the target table.
+     *
+     * @param string $key the key to be used to link both tables together
+     * @return $this
+     */
+    public function setTargetBindingKey($key)
+    {
+        $this->_targetBindingKey = $key;
+
+        return $this;
+    }
+
+    /**
+     * Gets the name of the field representing the binding key to the target table.
+     *
+     * @return array|string
+     */
+    public function getTargetBindingKey()
+    {
+        if ($this->_targetBindingKey === null) {
+            $this->_targetBindingKey = $this->isOwningSide($this->getSource()) ?
+                $this->getSource()->getPrimaryKey() :
+                $this->getTarget()->getPrimaryKey();
+        }
+
+        return $this->_targetBindingKey;
+    }
+
+    /**
+     * Sets the name of the field representing the binding key to the target table.
+     * If no parameters are passed current field is returned
+     *
+     * @deprecated 3.4.3 Use setTargetBindingKey()/getTargetBindingKey() instead.
+     * @param string|null $key the key to be used to link both tables together
+     * @return array|string|$this
+     */
+    public function targetBindingKey($key = null)
```

Comment:
Adding methods that are deprecated doesn't make sense to me. I would not include this method.